src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +153,1304 @@ static DamageTable D_80B65044 = {
+    }
+
+    this->actor.world.pos.y += this->actor.velocity.y;
+    func_80B619FC(this, globalCtx);
+    if (this->actor.velocity.y >= 4.0f) {
+        EnBigpo_SpawnPoCutscene5(this);
+    }
+}
+
+// fires have touched, they start to rise, big poe starts to visibly apear
+void EnBigpo_SpawnPoCutscene5(EnBigpo* this) {
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06001360);
+    this->actor.draw = EnBigpo_Draw1;
+    Actor_SetScale(&this->actor, 0.014f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALKIDS_APPEAR);
+    this->actionFunc = EnBigpo_SpawnPoCutscene6;
+}
+
+// big poe becoming visible in the flames
+void EnBigpo_SpawnPoCutscene6(EnBigpo* this, GlobalContext* globalCtx) {
+    s32 i;
+    s32 alphaPlus; // color alpha + 10
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->actor.shape.rot.y += this->rotVelocity;
+    alphaPlus = this->mainColor.a + 0xA; // decrease transparency
+    func_80B61914(this);
+    if (alphaPlus >= 0x5A) {
+        this->rotVelocity -= 0x80;
+        this->actor.velocity.y -= 0.25f;
+        if (alphaPlus >= 0xB4) {
+            Math_ScaledStepToS(&this->actor.world.rot.y, 0, 0x180);
+        }
+    }
+    this->actor.world.pos.y += this->actor.velocity.y;
+
+    for (i = 0; i < 3; i++) {
+        this->fires[i].pos.y += this->actor.velocity.y;
+    }
+
+    func_80B619FC(this, globalCtx);
+    if (alphaPlus >= 0xFF) {
+        this->mainColor.a = 0xFF; // fully visible
+        EnBigpo_SpawnPoCutscene7(this);
+    } else {
+        this->mainColor.a = alphaPlus;
+    }
+}
+
+// big poe is now fully visible
+void EnBigpo_SpawnPoCutscene7(EnBigpo* this) {
+    this->idleTimer = 15;
+    if (this->unkBool204 == false) {
+        func_801A2E54(0x38);
+        this->unkBool204 = true;
+    }
+    this->actionFunc = EnBigpo_SpawnPoCutscene8;
+}
+
+// count 15 frames, animating, then start dampe cutscene if hes here
+// if no dampe, camera pans back to player
+void EnBigpo_SpawnPoCutscene8(EnBigpo* this, GlobalContext* globalCtx) {
+    Actor* dampe;
+    Camera* cam;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->idleTimer -= 1;
+    if (this->idleTimer == 0) {
+        cam = Play_GetCamera(globalCtx, this->cutsceneCam);
+        func_8016970C(globalCtx, 0, &cam->at, &cam->eye);
+        this->cutsceneCam = NULL;
+        if (this->actor.params == ENBIGPO_SUMMONED) {
+            dampe = func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_NPC, ACTOR_EN_TK);
+            if (dampe != NULL) {
+                // if dampe exists, switch to viewing his running away cutscene
+                dampe->params = this->actor.cutscene;
+            } else {
+                ActorCutscene_Stop(this->actor.cutscene);
+            }
+        } else { // ENBIGPO_REGULAR
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+        func_800B724C(globalCtx, &this->actor, 6);
+        EnBigpo_SetupIdleFlying(this); // setup idle flying
+    }
+}
```

Comment:
```suggestion
// count 15 frames, animating, then start dampe cutscene if hes here
// also sets the main camera to align with the subCamera and switches back from the subCamera back to the main camera
void EnBigpo_SpawnPoCutscene8(EnBigpo* this, GlobalContext* globalCtx) {
    Actor* dampe;
    Camera* subCamId;

    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
    this->idleTimer -= 1;
    if (this->idleTimer == 0) {
        subCamId= Play_GetCamera(globalCtx, this->camId);
        func_8016970C(globalCtx, MAIN_CAM, &subCamId->at, &subCamId->eye);
        this->camId = MAIN_CAM;
        if (this->actor.params == ENBIGPO_SUMMONED) {
            dampe = func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_NPC, ACTOR_EN_TK);
            if (dampe != NULL) {
                // if dampe exists, switch to viewing his running away cutscene
                dampe->params = this->actor.cutscene;
            } else {
                ActorCutscene_Stop(this->actor.cutscene);
            }
        } else { // ENBIGPO_REGULAR
            ActorCutscene_Stop(this->actor.cutscene);
        }
        func_800B724C(globalCtx, &this->actor, 6);
        EnBigpo_SetupIdleFlying(this); // setup idle flying
    }
}
```

On second thought, since `this->camId` gets set back to the main camera, I think it should be called `this->camId`